pkgs/development/coq-modules/mathcomp-analysis/default.nix

```diff
@@ -25,4 +32,12 @@ with lib; mkCoqDerivation {
     maintainers = [ maintainers.cohencyril ];
     license = licenses.cecill-c;
   };
-}
+}; in
+mca.overrideAttrs (o:
+  let ext = { propagatedBuildInputs = o.propagatedBuildInputs
+                                      ++ [ hierarchy-builder ]; };
```

Comment:
Why don’t you put the correct buildInputs directly in the derivation, a few lines above?

Reply:
Because `hierarchy-builder` does not exist for some versions of Coq that are still supported by `mathcomp-analysis`

Reply:
What’s wrong with writing `propagatedBuildInputs = [ mathcomp.ssreflect … mathcomp-real-closed ] ++ optional (version == "dev" || versions.isGe "0.3.4" version) hierarchy-builder`?

Reply:
Because the computed version is not accessible at this stage, I should add an optional argument to `mkCoqDerivation` argument to get the current version.

Reply:
Got it. Thanks for the clarification. But the result is a bit ugly, isn’t it? Can’t we do better?

Reply:
Right now we cannot, but I will do a small refactoring of `mkCoqDerivation` soon-ish to make this more beautiful